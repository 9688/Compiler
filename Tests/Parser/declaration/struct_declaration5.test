
struct A{
	struct B{
		struct{} unnname;
	} b;
	B b2;
} a;

