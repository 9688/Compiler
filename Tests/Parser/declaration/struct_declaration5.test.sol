<<<table #1>>>
<tags>
A record
   <<<table #2>>>
   <tags>
   B record
      <<<table #3>>>
      <tags>
      <types>
      <vars>
      var s type  record
                    <<<table #4>>>
                    <tags>
                    <types>
                    <vars>
                    var x type int
                    var y type int
                  endrecord
    endrecord
   <types>
   <vars>
   var b type B record
               endrecord
   var b2 type B record
                endrecord
 endrecord
<types>
int
char
double
void
<vars>
var a type A record
            endrecord

