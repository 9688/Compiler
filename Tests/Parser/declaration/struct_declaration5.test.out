<--------TABLES-------->
<---TABLE--->
VARS::

a   STRUCT A{
vars::

b   STRUCT B{
vars::

unnname   STRUCT $UNNAMED${
vars::

types::
}   


types::
}   


b2   STRUCT B{
vars::

unnname   STRUCT $UNNAMED${
vars::

types::
}   


types::
}   


types::
}   


TYPES::
char char

double double

int int

void void

B STRUCT B{
vars::

unnname   STRUCT $UNNAMED${
vars::

types::
}   


types::
}

A STRUCT A{
vars::

b   STRUCT B{
vars::

unnname   STRUCT $UNNAMED${
vars::

types::
}   


types::
}   


b2   STRUCT B{
vars::

unnname   STRUCT $UNNAMED${
vars::

types::
}   


types::
}   


types::
}

